sct-commons/src/test/java/org/lfenergy/compas/sct/commons/testhelpers/ControlBlockNetworkingCsvHelper.java

```diff
@@ -0,0 +1,202 @@
+            ));
+    }
+
+    private Template findTemplate(Criteria criteria) {
+        Objects.requireNonNull(criteria);
+        if (criteria.controlBlockEnum() == null
+            || criteria.systemVersionWithoutV() == null
+            || criteria.iedType() == null
+            || criteria.iedRedundancy() == null) {
+            return null;
+        }
+        return templates.get(criteria);
+    }
+
+    private static String removeVFromSystemVersion(String systemVersion) {
+        if (systemVersion == null) {
+            return null;
+        }
+        String[] systemVersionParts = systemVersion.split("\\.");
+        if (systemVersionParts.length < 4) {
+            return systemVersion;
+        }
+        return systemVersionParts[0] + "." + systemVersionParts[1] + "." + systemVersionParts[2] + "." + systemVersionParts[3];
+    }
+
+    private static Criteria rowToCriteria(Row row) {
+        if (StringUtils.isBlank(row.cbType)
+            || StringUtils.isBlank(row.xy)
+            || StringUtils.isBlank(row.zw)
+            || StringUtils.isBlank(row.iedType)
+            || StringUtils.isBlank(row.bindingType)
+        ) {
+            throw new IllegalArgumentException("A least one criteria (cbType, xy, zw, iedType, bindingType) is blank");
+        }
+        ControlBlockEnum controlBlockEnum = switch (row.cbType) {
+            case "GOOSE" -> ControlBlockEnum.GSE;
+            case "SV" -> ControlBlockEnum.SAMPLED_VALUE;
+            default -> throw new IllegalArgumentException("Unsupported Control Block Type : " + row.cbType);
+        };
+        return new Criteria(
+            controlBlockEnum,
+            row.xy + "." + row.zw,
+            TCompasIEDType.fromValue(row.iedType),
+            TCompasIEDRedundancy.fromValue(row.iedRedundancy),
+            row.bindingType.equals("BAY_INTERNAL")
+        );
+    }
+
+    private static Template rowToTemplate(Row row) {
+        Integer vlanId = toVLanId(row.vlanId);
+        Byte vlanPriority = toVlanPriority(row.vlanPriority);
+        TDurationInMilliSec minTime = toDurationInMilliSec(row.minTime);
+        TDurationInMilliSec maxTime = toDurationInMilliSec(row.maxTime);
+        return new Template(vlanId, vlanPriority, minTime, maxTime);
+    }
+
+    @SuppressWarnings("WrapperTypeMayBePrimitive")
+    private static Byte toVlanPriority(String vlanPriority1) {
+        if (!NumberUtils.isDigits(vlanPriority1)) {
+            return null;
+        }
+        Byte vlanPriority = Byte.valueOf(vlanPriority1);
+        if (vlanPriority < 0 || vlanPriority > MAX_VLAN_PRIORITY) {
+            throw new IllegalArgumentException("VLAN PRIORITY must be between 0 and %d, but got : %d".formatted(MAX_VLAN_PRIORITY, vlanPriority));
+        }
+        return vlanPriority;
+    }
+
+    @SuppressWarnings("WrapperTypeMayBePrimitive")
+    private static Integer toVLanId(String vlanId1) {
+        if (!NumberUtils.isDigits(vlanId1)) {
+            return null;
```

Comment:
Why not throw exception? it could help to return primitive and delete warning.

Reply:
same comment as toVlanPriority.

Reply:
ok for your proposal

Reply:
OK